raster/apply-mosaic-rule-to-rasters/src/main/java/com/esri/samples/apply_mosaic_rule_to_rasters/ApplyMosaicRuleToRastersSample.java

```diff
@@ -0,0 +1,214 @@
+
+      // set the map to the map view
+      mapView.setMap(map);
+
+      // create a raster layer from the image service raster
+      ImageServiceRaster imageServiceRaster = new ImageServiceRaster("https://sampleserver7.arcgisonline.com/server/rest/services/amberg_germany/ImageServer");
+      rasterLayer = new RasterLayer(imageServiceRaster);
+
+      // add a raster layer as an operational layer to the map
+      map.getOperationalLayers().add(rasterLayer);
+
+      // wait for the raster layer to finish loading
+      rasterLayer.addDoneLoadingListener(() -> {
+        if (rasterLayer.getLoadStatus() == LoadStatus.LOADED) {
+          // when loaded, set the map view's viewpoint to the image service raster's center
+          mapView.setViewpoint(new Viewpoint(imageServiceRaster.getServiceInfo().getFullExtent().getCenter(), 25000.0));
+
+          // enable UI interaction once the raster layer has loaded
+          controlsVBox.setVisible(true);
+
+          // create a mosaic rule for the image service raster if one does not already exist
+          if (imageServiceRaster.getMosaicRule() == null) {
+            mosaicRule = new MosaicRule();
+            imageServiceRaster.setMosaicRule(mosaicRule);
+          }
+        } else {
+          // show alert if raster layer fails to load.
+          new Alert(Alert.AlertType.ERROR, "Error loading raster layer.").show();
+        }
+      });
+
+      // set the mosaic rule of the image service raster based on rule chosen from the combo box
+      mosaicRuleComboBox.getSelectionModel().selectedItemProperty().addListener(e -> {
+        // create a new mosaic rule
+        mosaicRule = new MosaicRule();
+
+        switch (mosaicRuleComboBox.getSelectionModel().getSelectedItem()) {
+          case "Default":
+            mosaicRule.setMosaicMethod(MosaicMethod.NONE);
+            break;
+          case "Northwest":
+            mosaicRule.setMosaicMethod(MosaicMethod.NORTHWEST);
+            mosaicRule.setMosaicOperation(MosaicOperation.FIRST);
+            break;
+          case "Center":
+            mosaicRule.setMosaicMethod(MosaicMethod.CENTER);
+            mosaicRule.setMosaicOperation(MosaicOperation.BLEND);
+            break;
+          case "By attribute":
+            mosaicRule.setMosaicMethod(MosaicMethod.ATTRIBUTE);
+            mosaicRule.setSortField("OBJECTID");
+            break;
+          case "Lock raster":
+            mosaicRule.setMosaicMethod(MosaicMethod.LOCK_RASTER);
+            mosaicRule.getLockRasterIds().clear();
+            mosaicRule.getLockRasterIds().addAll(Arrays.asList(1L, 7L, 12L));
+            break;
+        }
+        // set the mosaic rule of the image service raster
+        imageServiceRaster.setMosaicRule(mosaicRule);
+      });
+
+      // add the map view, control panel and progress indicator to the stack pane
+      stackPane.getChildren().addAll(mapView,controlsVBox, progressIndicator);
```

Comment:
Missing a space here:
```suggestion
      stackPane.getChildren().addAll(mapView, controlsVBox, progressIndicator);
```